tests/scripts/check-all_sh-components.sh

```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/env sh
+
+# This file is part of mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2018, ARM Limited, All Rights Reserved
+#
+# Purpose
+#
+# Check if the list of components in all.sh is up-to-date
+
+set -eu
+
+DEFINED="def.$$"
+LISTED="lst.$$"
+
+sed -n 's/^component_\([^ (]*\).*/\1/p' tests/scripts/all.sh \
```

Comment:
I agree with this approach of looking for definitions of functions whose name starts with `component_`, but I think the regex should allow indentation, and require `()` after the function name.
```
s/^ *component_\([0-9A-Z_a-z]*\) *().*/\1/p
```